packages/eslint-plugin/tests/rules/prefer-nullish-coalescing.test.ts

```diff
@@ -751,5 +759,129 @@ declare const c: ${type};
         },
       ],
     })),
+    ...ignorablePrimitiveTypes.map<
+      TSESLint.InvalidTestCase<MessageIds, Options>
+    >(ignoreablePrimitive => ({
+      code: `
+declare const x: ${ignoreablePrimitive} | undefined;
+x || y;
+      `,
+      options: [
+        {
+          ignorePrimitives: Object.fromEntries(
+            ignorablePrimitiveTypes
+              .filter(t => t !== ignoreablePrimitive)
+              .map(t => [t, true]),
+          ),
+        },
+      ],
+      errors: [
+        {
+          messageId: 'preferNullishOverOr',
+        },
+      ],
+    })),
+    ...[
+      // falsy
+      { ignoreablePrimitive: ['string'], literalPrimitive: "''" },
+      { ignoreablePrimitive: ['string'], literalPrimitive: '``' },
+      { ignoreablePrimitive: ['number'], literalPrimitive: '0' },
+      { ignoreablePrimitive: ['bigint'], literalPrimitive: '0n' },
+      { ignoreablePrimitive: ['boolean'], literalPrimitive: 'false' },
+      // truthy
+      { ignoreablePrimitive: ['string'], literalPrimitive: "'a'" },
+      {
+        ignoreablePrimitive: ['string'],
+        literalPrimitive: "`hello${'string'}`",
+      },
+      { ignoreablePrimitive: ['number'], literalPrimitive: '1' },
+      { ignoreablePrimitive: ['bigint'], literalPrimitive: '1n' },
+      { ignoreablePrimitive: ['boolean'], literalPrimitive: 'true' },
+      // unions
+      { ignoreablePrimitive: ['string'], literalPrimitive: "'a' | 'b'" },
+      { ignoreablePrimitive: ['string'], literalPrimitive: "'a' | `b`" },
+      { ignoreablePrimitive: ['number'], literalPrimitive: '0 | 1' },
+      { ignoreablePrimitive: ['number'], literalPrimitive: '1 | 2 | 3' },
+      { ignoreablePrimitive: ['bigint'], literalPrimitive: '0n | 1n' },
+      { ignoreablePrimitive: ['bigint'], literalPrimitive: '1n | 2n | 3n' },
+      { ignoreablePrimitive: ['boolean'], literalPrimitive: 'true | false' },
+      // Mixed unions
+      {
+        ignoreablePrimitive: ['number'],
+        literalPrimitive: '0 | 1 | 0n | 1n',
+      },
+      {
+        ignoreablePrimitive: ['bigint'],
+        literalPrimitive: '0 | 1 | 0n | 1n',
+      },
+      {
+        ignoreablePrimitive: ['number', 'bigint'],
+        literalPrimitive: '0 | 1 | 0n | 1n',
+      },
+      {
+        ignoreablePrimitive: ['boolean'],
+        literalPrimitive: 'true | false | null',
+      },
+    ].map<TSESLint.InvalidTestCase<MessageIds, Options>>(
```

Comment:
> `.map`

[Testing] I can't find a tracking right now, but I'm slowly getting more and more sour on generating tests this way. I'm finding it very difficult to read through them and the generation code obfuscates what the test actually does. Could you just have manually written error cases for these please?

Reply:
I understand what you mean here, I'm all for flatting them, I thought it was how it works here based on a small sample of tests I've seen